Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/DefaultPointerMediator.cs

```diff
@@ -167,6 +171,49 @@ public virtual void UpdatePointers()
             {
                 unassignedPointer.IsActive = true;
             }
+
+        }
+
+        private void ApplyCustomPointerBehaviors()
+        {
+            if (CoreServices.InputSystem.FocusProvider is FocusProvider focusProvider)
+            {
+                Action<IMixedRealityPointer, PointerBehavior> setPointerState =
+                    (ptr, behavior) =>
+                    {
+                        if (behavior == PointerBehavior.Off)
+                        {
+                            ptr.IsActive = false;
+                            if (ptr is GenericPointer genericPtr)
+                            {
+                                genericPtr.IsInteractionEnabled = false;
+                            }
+                            unassignedPointers.Remove(ptr);
+                        }
+                        else if (behavior == PointerBehavior.On)
+                        {
+                            ptr.IsActive = true;
+                            if (ptr is GenericPointer genericPtr)
+                            {
+                                genericPtr.IsInteractionEnabled = true;
+                            }
+                            unassignedPointers.Remove(ptr);
+                        }
+                    };
+                foreach (IMixedRealityPointer pointer in allPointers)
+                {
+                    Handedness handedness = pointer.Controller == null ? Handedness.None : pointer.Controller.ControllerHandedness;
+                    if (pointer is GGVPointer ||
+                        pointer == CoreServices.InputSystem.GazeProvider.GazePointer)
+                    {
+                        setPointerState(pointer, focusProvider.GazePointerBehavior);
```

Comment:
If we're going to have a class that gets pointer behavior, why not abstract away the complication of specific pointer interactions behind that? i.e. it feels like this should be in GetPointerBehavior, which should work for all pointers. Otherwise, would the utils also need to handle this specially as well?

Reply:
Ah no, the utils do exactly the same thing - I would still say that we should have a single implementation of this, otherwise if focusProvider.GetPointerBehavior is not safe to call by itself it would open us to new bugs in the future where we forget to special case the gaze pointer.

Reply:
So if I'm understanding, you want a way to get the pointer behavior for the gaze pointer.

How would you make that call when the gaze pointer is an internal class in GazeProvider?

I guess we could say "if you want to get the gaze pointer behavior, you need to instead actually call GetPointerBehavior<GGVPointer>"

Reply:
Ah, I think the internal gaze pointer thing is what complicates things...

I think my expectation here is that the gaze pointer would have been like other pointers, but because it's internal I guess it doesn't work out.